sdks/go/README.md

```diff
@@ -84,56 +145,237 @@ sentence: 1
+   for more details):
+    ```
+    go run go/examples/wordcount/wordcount.go --runner=dataflow \
+        --sdk_container_image=apache/beam_go_sdk:latest \
+        --project=$GCP_PROJECT \
+        --region=$GCP_REGION \
+        --staging_location=gs://$GCS_BUCKET/staging \
+        --output=gs://$GCS_BUCKET/output
+    ```
+
+    You should see:
+    ```
+    2023/07/09 10:57:13 Submitted job: <job-id>
+    2023/07/09 10:57:13 Console: https://console.cloud.google.com/dataflow/jobs/us-central1/<job-id>?project=<project>
+    2023/07/09 11:02:56 Job state: JOB_STATE_PENDING ...
+    2023/07/09 11:03:26 Job still running ...
+    ```
+
+5. After seeing `Job <job-id> succeeded!` you can inspect the resulting output.
+
+    Run the following command to inspect the resulting output.
+
+    ```
+    gcloud storage cat "gs://$GCS_BUCKET/output*" | head
+    ```
+
+    You should see something similar to the following:
+    ```
+    feature: 1
+    block: 1
+    Cried: 1
+    scatter'd: 1
+    she: 44
+    sudden: 1
+    silly: 1
+    More: 6
+    out: 68
+    believe: 3
+    ```
+
+#### Troubleshooting tips
+
+If you get the following error:
+```
+googleapi: Error 400: User project specified in the request is invalid.
+```
+
+Try the following:
+1. Make sure you followed the [Google Cloud Setup](#google-cloud-setup) above.
+2. Configure the [gcloud](https://cloud.google.com/sdk/docs/install-sdk) with your project:
+   ```
+   gcloud config set project <your-project>
+   ```
+3. Consider re-running BOTH `gcloud auth login` and `gcloud auth application-default login`
+   commands.
+
+## Testing
+
+### Requirements (For runner validations)
+
+Below lists additional requirements to execute tests in this repository on your local machine.
+
+#### 1. Java and Python
+
+You **do not** need to know or care about Java or Python to use the Go SDK for your data processing goals.
+
+Java is **only** required to execute any [gradle](https://gradle.org/) commands configured in the
+[Beam](https://github.com/apache/beam) repository. It will be obvious whether you will execute gradle commands in
+sections below. You **do not** need to install [gradle](https://gradle.org/) and simply use the enclosed
+`$BEAM_ROOT/gradlew` executable available at the root of the [Beam](https://github.com/apache/beam) repository.
+
+Python is **only** required if you need to validate tests against the
+[Portable Python Runner](#portable-python-runner). If you are not testing against this
+runner, ignore the Python requirement.
+
+#### 2. Docker
+
+[Docker](https://www.docker.com/) is required for certain but not all runners [See definition](#runner).
+
+As of this writing, [colima](https://github.com/abiosoft/colima), a preferred docker alternative for some developers,
+did not work.
+
+#### 3. Flock
+
+[sdks/go/run_with_go_version.sh](run_with_go_version.sh) requires the use of
```

Comment:
This isn't necessary to mention. The gradle commands are largely for the github actions or the java developers. The reason flock exists is to avoid race conditions on installing Go versions, and once prism is the default, users won't need to run these scripts locally, they should use Go test.